app/goal_routes.py

```diff
@@ -0,0 +1,156 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, abort, make_response, request, jsonify
+from sqlalchemy import select
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+def validate_goal(goal_id):
+    try:
+        goal_id = int(goal_id)
+    except:
+        abort(make_response({"message": f"goal {goal_id} is invalid"}, 400))
+    
+    goal = Goal.query.get(goal_id)
+    
+    if not goal:
+        abort(make_response({"details": f"Goal {goal_id} not found"}, 404))
+    
+    return goal
+
+def validate_task_list(task_ids):
+    try:
+        task_ids = list(task_ids)
+    except:
+        abort(make_response({"message": f"task_ids list {task_ids} is invalid"}, 400))
+    
+    task_list = []
+    
+    for task_id in task_ids:
+        task = Task.query.get(task_id)
+        if not task:
+            abort(make_response({"details": f"Task {task_id} not found"}))
+        task_list.append(task)
+    
+    return task_list
+
+# def validate_task(task_id):
+#     try:
+#         task_id = int(task_id)
+#     except:
+#         abort(make_response({"message": f"task {task_id} is invalid"}, 400))
+
+#     task = Task.query.get(task_id)
+    
+#     if not task:
+#         abort(make_response({"details": f"Task {task_id} not found"}, 404))
+    
+#     return task
+    
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    
+    if "title" not in request_body:
+        abort(make_response({"details": "Invalid data"}, 400))
+    
+    new_goal = Goal(title=request_body["title"])
+    
+    db.session.add(new_goal)
+    db.session.commit()
+    
+    return {
```

Comment:
Here you could have used the `to_dict()` instance method defined in the Goal model.